presidio-api/cmd/presidio-api/api/stream-job/stream-job.go

```diff
@@ -0,0 +1,89 @@
+package streamjob
+
+import (
+	"context"
+	"fmt"
+
+	types "github.com/Microsoft/presidio-genproto/golang"
+	"github.com/Microsoft/presidio/pkg/presidio"
+	store "github.com/Microsoft/presidio/presidio-api/cmd/presidio-api/api"
+	"github.com/Microsoft/presidio/presidio-api/cmd/presidio-api/api/templates"
+)
+
+//ScheduleStreamsJob schedule stream job
+func ScheduleStreamsJob(ctx context.Context, api *store.API, streamsJobAPIRequest *types.StreamsJobApiRequest, project string) (*types.StreamsJobResponse, error) {
+
+	streamsJobRequest, err := getStreamsJobRequest(api, streamsJobAPIRequest, project)
+	if err != nil {
+		return nil, err
+	}
+	schedulerResponse, err := invokeStreamsJobScheduler(ctx, api.Services, streamsJobRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return schedulerResponse, nil
+}
+
+func invokeStreamsJobScheduler(ctx context.Context, services presidio.ServicesAPI, streamsJobRequest *types.StreamsJobRequest) (*types.StreamsJobResponse, error) {
+	res, err := services.ApplyStream(ctx, streamsJobRequest)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+func getStreamsJobRequest(api *store.API, jobAPIRequest *types.StreamsJobApiRequest, project string) (*types.StreamsJobRequest, error) {
+	streamsJobRequest := &types.StreamsJobRequest{}
+
+	if jobAPIRequest.GetStreamsJobTemplateId() != "" {
```

Comment:
add a validate function

Reply:
Will do in a separate PR for scanner and streams with other features